### PR TITLE
[risk=no] Remove Angular codegen

### PR DESCRIPTION
**Pull request by calbach**
A couple of changes here:

- Update some stale + incorrect usage of Angular interfaces (workspace-share, admin pages)
- Switch remaining angular service usage (from internal services/guards) to fetch
- Switch existing stores/services to use fetch models
- Remove dead test stubs
- Refactor bootstrapping of fetch APIs to cover `ConfigApi`
- Remove Angular codegen

### CODE REVIEW

File: ui/src/app/utils/index.tsx [hunk 1]
```diff
@@ -45,21 +47,13 @@ export function isBlank(toTest: String): boolean {
  * minimum required level to do most things in the Workbench app (outside of
  * local/test development).
  */
-export function hasRegisteredAccess(access: DataAccessLevel): boolean {
+export function hasRegisteredAccessFetch(access: DataAccessLevel): boolean {
```
Comment (als364) [inline ui/src/app/utils/index.tsx:50]: nit: this doesn't need to be called `hasRegisteredAccessFetch` anymore in the absence of `hasRegisteredAccess`

File: ui/src/app/utils/stores.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { BreadcrumbType } from 'app/utils/navigation';
 import {atom, Atom} from 'app/utils/subscribable';
-import {Profile} from 'generated';
+import {Profile} from 'generated/fetch';
```
Comment (als364) [inline ui/src/app/utils/stores.tsx:3]: nit: collapse into line below (think this should be covered by lint)

File: ui/src/app/services/profile-storage.service.ts [hunk 3]
```diff
@@ -49,17 +49,13 @@ export class ProfileStorageService {
     }
     this.activeCall = true;
 
-    // Conflict 409s are not retries automatically. These have a likelihood to
```
Comment (als364) [inline ui/src/app/services/profile-storage.service.ts:52]: Is this comment no longer true with your changes?

File: ui/src/app/guards/workspace-guard.service.ts [hunk 4]
```diff
@@ -3,22 +3,21 @@ import {
...
+  constructor(private router: Router) {}
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
     if (serverConfigStore.getValue().enableResearchReviewPrompt && route.routeConfig.path === 'data' ||
         route.routeConfig.path === 'notebooks') {
-      this.workspaceService.getWorkspace(route.params.ns, route.params.wsid).subscribe(workspace => {
-        if (workspace.accessLevel === 'OWNER'
-            && workspace.workspace.researchPurpose.needsReviewPrompt) {
+      workspacesApi().getWorkspace(route.params.ns, route.params.wsid).then((resp: WorkspaceResponse) => {
+        if (resp.accessLevel === WorkspaceAccessLevel.OWNER
```
Comment (als364) [inline ui/src/app/guards/workspace-guard.service.ts:19]: nice catch

File: ui/src/app/services/profile-storage.service.ts (hunk 3 above)
Comment (calbach) [reply ui/src/app/services/profile-storage.service.ts:52]: A few parts to this one:

1. The rationale for this behavior is a bit outdated. In the past, we did a lot more lazy initialization with the profile API, and the client made many more parallel requests, so we saw lots of 409s on first login. We've untangled a lot of that logic, and this doesn't really happen anymore. I don't see this at all in test, and in prod in the past 21 days, there's only been one case of a 409 on the profile API (and this retry didn't help - it just 409'd 3 times in a row).
2. It is still true that 409s are not retried automatically (actually, with the fetch client, even 500s aren't being retried automatically, which is something we will want to address).
3. With this change, we're removing the 409 retries, which weren't actively bad, but also are no longer necessary. If we had a drop-in replacement, I might put it here.

File: ui/src/app/utils/stores.tsx (hunk 2 above)
Comment (calbach) [reply ui/src/app/utils/stores.tsx]: FYI: lint didn't catch - but fixed manually

**Review by als364 — APPROVED** [review]
Oop, thought I had pushed the button and I had not. LGTM

**Pull request merged**